### DISCUSSION TITLE
Jackson Serialization

### DISCUSSION THREAD

**Discussion by owengalenjones** [graphql-java-kickstart/graphql-spring-boot#981] [Q&A]
Hello, I'm a little unclear about how to go about using Jackson to serialize objects. I have an annotated POJO:

```java

public class Foo {

@JsonProperty("bar")
private String something;

// getters / setters
}
```

and am invoking GraphQL through the DGS framework:

```java
return executor.execute(
        body.getQuery(),
        body.getVariables(),
        body.getOperationName(),
    );
```

The `data` property that I get back is an `Object` typed `LinkedHashMap` that does _not_ appear to have followed the Jackson annotations:

```
ExecutionResultImpl{errors=[], data={library={something=bad}}, dataPresent=true, extensions=null}
data LinkedHashMap
  "foo" LinkedHashMap
    key = "foo"
    value = "something" -> "bad"
```

It appears that something has per-emptively changed the type to `LinkedHashMaps` prior to running Jackson serialization, resulting in the actual field names, not the annotated JSON properties.

If I manually serialize it I get the expected result (the property obeys the `@JsonProperty` annotation):

```
new ObjectMapper().writeValueAsString(foo);
=> {"bar": "good"}
```

I can't tell if this is a graphql-java or DGS issue, apologies if it's the later!

**Comment by oliemansm** [comment]
In both cases this isn't the right project to ask, since this is the kickstart library that uses the graphql-java lib under the hood. I suggest you first check out the DGS project and get support there. Could be they'll refer you to graphql-java if necessary.
